File: src/easy/number_of_steps_to_zero.zig
const std = @import("std");

const expectEqual = std.testing.expectEqual;

pub fn numberOfStepsToZero(num: i32) usize {
    if (num < 0) return 0;
    var steps: usize = 0;
    var n = num;

    while (n > 0) : (steps += 1) {
        if (@mod(n, 2) == 0) {
            n = @divFloor(n, 2);
        } else {
            n -= 1;
        }
    }
    return steps;
}

File: src/easy/number_of_steps_to_zero.rs
pub fn number_of_steps_to_zero(num: i32) -> i32 {
    if num < 0 {
        return 0;
    }
    let mut n = num;
    let mut steps = 0;
    while n > 0 {
        if n % 2 == 0 {
            n /= 2;
        } else {
            n -= 1;
        }
        steps += 1;
    }
    steps
}
	